utils/docker-utils.sh

```diff
@@ -481,6 +481,37 @@ docker::ensure_external_networks_existence() {
     done
 }
 
+# Joins a service to a network by updating the service spec to include the network.
+#
+# Note: Do not remove if not used in the Platform as this is mainly used by
+# custom packages that cannot overwrite the docker compose file to add the network connection required.
+#
+# Arguments:
+# - $1 : service name that needs to join the network (eg. analytics-datastore-elastic-search)
+# - $2 : network name to join (eg. elastic_public)
+#
+docker::join_network() {
+    local -r SERVICE_NAME="${1:?$(missing_param "join_network" "SERVICE_NAME")}"
+    local -r NETWORK_NAME="${2:?$(missing_param "join_network" "NETWORK_NAME")}"
+    local network_id
+    network_id=$(docker network ls --filter name="$NETWORK_NAME" --format '{{.ID}}')
```

Comment:
You probably want to regex match the exact network name instead, just to future proof any scenario where we might have a network `foo_public` and `foo_public_backup`. Which will currently pull out both. `$NETWORK_NAME$` should be enough, or you can be even more restrictive with `^$NETWORK_NAME$`.